MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockMpService.h

```diff
@@ -0,0 +1,124 @@
+
+extern "C" {
+  #include <Uefi.h>
+  #include <Protocol/MpService.h>
+}
+
+struct MockMpService {
+  MOCK_INTERFACE_DECLARATION (MockMpService);
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    GetNumberOfProcessors,
+    (
+     IN  EFI_MP_SERVICES_PROTOCOL  *This,
+     OUT UINTN                     *NumberOfProcessors,
+     OUT UINTN                     *NumberOfEnabledProcessors
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    GetProcessorInfo,
+    (
+     IN  EFI_MP_SERVICES_PROTOCOL   *This,
+     IN  UINTN                      ProcessorNumber,
+     OUT EFI_PROCESSOR_INFORMATION  *ProcessorInfoBuffer
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    StartupAllAPs,
+    (
+     IN  EFI_MP_SERVICES_PROTOCOL  *This,
+     IN  EFI_AP_PROCEDURE          Procedure,
+     IN  BOOLEAN                   SingleThread,
+     IN  EFI_EVENT                 WaitEvent               OPTIONAL,
+     IN  UINTN                     TimeoutInMicroSeconds,
+     IN  VOID                      *ProcedureArgument      OPTIONAL,
+     OUT UINTN                     **FailedCpuList         OPTIONAL
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    StartupThisAP,
+    (
+     IN  EFI_MP_SERVICES_PROTOCOL  *This,
+     IN  EFI_AP_PROCEDURE          Procedure,
+     IN  UINTN                     ProcessorNumber,
+     IN  EFI_EVENT                 WaitEvent               OPTIONAL,
+     IN  UINTN                     TimeoutInMicroseconds,
+     IN  VOID                      *ProcedureArgument      OPTIONAL,
+     OUT BOOLEAN                   *Finished               OPTIONAL
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    SwitchBSP,
+    (
+     IN EFI_MP_SERVICES_PROTOCOL  *This,
+     IN  UINTN                    ProcessorNumber,
+     IN  BOOLEAN                  EnableOldBSP
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    EnableDisableAP,
+    (
+     IN  EFI_MP_SERVICES_PROTOCOL  *This,
+     IN  UINTN                     ProcessorNumber,
+     IN  BOOLEAN                   EnableAP,
+     IN  UINT32                    *HealthFlag OPTIONAL
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    WhoAmI,
+    (
+     IN EFI_MP_SERVICES_PROTOCOL  *This,
+     OUT UINTN                    *ProcessorNumber
+    )
+    );
+};
+
+MOCK_INTERFACE_DEFINITION (MockMpService);
+MOCK_FUNCTION_DEFINITION (MockMpService, GetNumberOfProcessors, 3, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockMpService, GetProcessorInfo, 3, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockMpService, StartupAllAPs, 7, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockMpService, StartupThisAP, 7, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockMpService, SwitchBSP, 3, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockMpService, EnableDisableAP, 4, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockMpService, WhoAmI, 2, EFIAPI);
+
+EFI_MP_SERVICES_PROTOCOL  MP_SERVICE_PROTOCOL_INSTANCE = {

```

Comment:
Can this struct be "static" since it is used within this same file?

Reply:
Done.